js/ui/components/clubhouse.js

```diff
@@ -77,6 +156,14 @@ var ClubhouseNotificationBanner = new Lang.Class({
 
         // Always wrap the body's text
         this._expandedLabel.actor.add_style_class_name('clubhouse-quest-view-label');
+
+        this._clubhouseWindowTracker = new ClubhouseWindowTracker();
```

Comment:
I'm concerned that we create one of these for each banner (plus one for the button manager), but they will never get destroyed. There are a few problems with that:
- these objects are not cheap, as they react to frequent signals (`restacked`) and iterate over all the windows.
- I know that the banner disconnects the signal when it's destroyed, but each tracker object itself will stay alive and keep all of its connections/signals alive and running. That may be avoidable by setting `this._clubhouseWindowTracker = null` after disconnecting the signals. It should get garbage-collected.
- If you do the above and manage to destroy the tracker, the callbacks from the overview and sessionMode will still be connected, since those objects outlive the tracker and nothing disconnects the signals when the tracker is destroyed. I think that at present, those callbacks would crash the app in that situation, since the code would still try to call `emit()` on an instance that is already destroyed.

I am open to either solving this by creating a single tracker object, owned by the clubhouse component, that ends up being shared by all the other objects that need it (my preferred solution), or by making sure to destroy the trackers with each banner, and making the tracker robust to that case (my less preferred solution because of the performance/resources concern.)